# ftp://invisible-island.net/temp/xterm-278-p4.patch.gz
# patch by Thomas E. Dickey <dickey@invisible-island.net>
# created  Tue Nov 27 10:17:35 UTC 2012
# ------------------------------------------------------------------------------
# charproc.c |    8 ++++----
# input.c    |    2 +-
# misc.c     |    4 ++--
# 3 files changed, 7 insertions(+), 7 deletions(-)
# ------------------------------------------------------------------------------
Index: charproc.c
--- xterm-278+/charproc.c	2012-01-19 00:22:08.000000000 +0000
+++ xterm-278-p4/charproc.c	2012-11-27 10:15:42.000000000 +0000
@@ -3797,7 +3797,7 @@
     XPoint spot;
     XVaNestedList list;
 
-    if (input->xic
+    if (input && input->xic
 	&& (ld = getLineData(screen, screen->cur_row)) != 0) {
 	spot.x = (short) LineCursorX(screen, ld, screen->cur_col);
 	spot.y = (short) (CursorY(screen, screen->cur_row) + xw->misc.xim_fs_ascent);
@@ -6952,7 +6952,7 @@
 {
     TInput *input = lookupTInput(xw, (Widget) xw);
 
-    if (input->xim) {
+    if (input && input->xim) {
 	XCloseIM(input->xim);
 	input->xim = 0;
 	TRACE(("freed screen->xim\n"));
@@ -7506,8 +7506,8 @@
 
     TRACE(("xim_destroy_cb im=%lx, client=%p, call=%p\n",
 	   (long) im, client_data, call_data));
-
-    input->xic = NULL;
+    if (input)
+	input->xic = NULL;
     XRegisterIMInstantiateCallback(XtDisplay(xw), NULL, NULL, NULL,
 				   xim_instantiate_cb, NULL);
 }
Index: input.c
--- xterm-278+/input.c	2012-01-07 02:01:05.000000000 +0000
+++ xterm-278-p4/input.c	2012-11-27 10:15:42.000000000 +0000
@@ -832,7 +832,7 @@
     {
 #if OPT_I18N_SUPPORT && OPT_INPUT_METHOD
 	TInput *input = lookupTInput(xw, (Widget) xw);
-	if (input->xic) {
+	if (input && input->xic) {
 	    Status status_return;
 #if OPT_WIDE_CHARS
 	    if (screen->utf8_mode) {
Index: misc.c
--- xterm-278+/misc.c	2012-01-07 01:57:52.000000000 +0000
+++ xterm-278-p4/misc.c	2012-11-27 10:15:42.000000000 +0000
@@ -180,7 +180,7 @@
     {
 #if OPT_I18N_SUPPORT && OPT_INPUT_METHOD
 	TInput *input = lookupTInput(xw, (Widget) xw);
-	if (input->xic)
+	if (input && input->xic)
 	    XSetICFocus(input->xic);
 #endif
 
@@ -218,7 +218,7 @@
 	{
 #if OPT_I18N_SUPPORT && OPT_INPUT_METHOD
 	    TInput *input = lookupTInput(xw, (Widget) xw);
-	    if (input->xic)
+	    if (input && input->xic)
 		XUnsetICFocus(input->xic);
 #endif
 
